sklearn/mixture/_gaussian_mixture.py

```diff
@@ -712,10 +725,12 @@ def _initialize(self, X, resp):
         """
         n_samples, _ = X.shape
 
```

Comment:
Currently it seems like weights, means and covariances have no meaning if we are outside the if condition which might not really be an issue in this case but it still seems a bit ambiguous. How about initialising them here or returning three None values from within the _estimate_gaussian_parameters function if resp is None.

```suggestion
        weights, means, covariances = None, None, None
```

Reply:
Thank you so much for your suggestion!